src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -920,7 +905,7 @@ private boolean canBeReferencedFromStaticContext(DetailAST ident) {
      *         definition was not found.
      */
     private static DetailAST getCodeBlockDefinitionToken(DetailAST ident) {
-        DetailAST parent = ident.getParent();
```

Comment:
The AST given is always `IDENT`.

This is the same thing we have seen everywhere. We just do an extra `getParent` to get back to where we would be without this mutation.